### PR TITLE
Read all vehicle models loaded into a 2D array on startup

### PR DESCRIPTION
**Pull request by zorg93**
The array can be accessed using Vehicle.GetAllModels() and Vehicle.GetAllModelsOfClass(VehicleClass vClass)
Idea behind this is so any mod that needs to use all vehicle list(maybe for car spawning) can just grab a full list, doesn't have to worry about addon cars or updating their mod when dlc gets released

### CODE REVIEW

**Comment by kagikn** [comment]
Would you find the pattern of ped models pool? If we find it, we can make `Model.IsPed` strict.

File: source/core/NativeMemory.cpp [hunk 1]
```diff
@@ -291,6 +292,64 @@ namespace GTA
...
+					UINT16 data = cur->data;
+					if ((int)data < cmp1 && bittest(*reinterpret_cast<int*>(mov2 + (4 * data >> 5)), data & 0x1F))
+					{
+						UINT64 addr1 = add4 + mul3 * data;
+						if (addr1)
+						{
+							UINT64 addr2 = *reinterpret_cast<PUINT64>(addr1);
+							if (addr2)
+							{
+								if ((*reinterpret_cast<PBYTE>(addr2 + 157) & 0x1F) == 5)
```
Comment (zorg93) [inline source/core/NativeMemory.cpp]: @kagikn To get peds just change that == 5 to == 6. I could improve this code a bit and make a function to check if a model is a ped though

**Comment by kagikn** [comment]
@zorg93 So, you can also make a function to detect if a model is a prop, right?

**Comment by zorg93** [comment]
@kagikn Yes, Just rewriting the original code to make it easier to work with getting peds/props

**Comment by zorg93** [comment]
Got peds working, now just need to do Props, shouldnt be too hard though :)

**Comment by zorg93** [comment]
@kagikn scratch that, props wont be possible as you can spawn vehicles (and peds?) as a prop, best bet would be to check if model is valid and not a ped or a vehicle

**Comment by Guad** [comment]
@zorg93 You can check if the prop is also a vehicle, if so exclude it :pray:

**Comment by zorg93** [comment]
@Guad IsProp = IsValid && !IsPed && !IsVehicle is sufficient. Im not going to try and generate lists of props or peds as they are just too large. You cant extract a name from the hash unlike with vehicles so any more work would be unnecessary.

**Comment by Guad** [comment]
That's a shame, but a vehicle list is still really good.

**Pull request merged**